src/bindings/python/flux/job/info.py

```diff
@@ -243,23 +243,48 @@ def convert_field(self, value, conv):
             different conversion types. (mainly used for time-specific
             fields for now).
             """
+            orig_value = str(value)
             if conv == "d":
                 # convert from float seconds since epoch to a datetime.
                 # User can than use datetime specific format fields, e.g.
                 # {t_inactive!D:%H:%M:S}.
-                value = datetime.fromtimestamp(value)
+                try:
+                    value = datetime.fromtimestamp(value)
+                except TypeError:
+                    if orig_value is "":
+                        value = datetime.fromtimestamp(0)
+                    else:
+                        raise
             elif conv == "D":
                 # As above, but convert to ISO 8601 date time string.
-                value = datetime.fromtimestamp(value).strftime("%FT%T")
+                try:
+                    value = datetime.fromtimestamp(value).strftime("%FT%T")
+                except TypeError:
+                    if orig_value is "":
+                        value = datetime.fromtimestamp(0).strftime("%FT%T")
```

Comment:
@chu11: without actually running this, it is kind of difficult to see how this behaves. 

If a job doesn't have `sched.t_estimate` which would be the case for the jobs that have never been reserved, will this print a very old date? like 1969? Or is this for something else?

Reply:
yeah, it'll print `1970-01-01`.  Which now that you mention it, may not be what we want.  I thought you could use the hyphen notation to solve this, but it seems datetime adjustments make it so it won't always work.

```
>flux jobs -A -no '{id},{annotations.sched.t_estimate!D:h}'
f4doYt4o,1969-12-31T16:00:00

>flux jobs -A -no '{id},{annotations.sched.t_estimate!F:h}'
f4doYt4o,-

>flux jobs -A -no '{id},{annotations.sched.t_estimate!d:h}'
f4doYt4o,1969-12-31 16:00:00

>flux jobs -A -no '{id},{annotations.sched.t_estimate!H:h}'
f4doYt4o,-
```

hmmm, would the following be preferred?

```
try:
        value = datetime.fromtimestamp(value)
except TypeError:
        if orig_value is "":
               value = ""
       else:
               raise
```

Reply:
Yeah, that might be better for conversion `D`, but for `d` you'll have to choose a floating point value.

Reply:
not sure what you mean by "for `d` you'll have to choose a floating point value"?  Do you output a literal `0.0` instead of `""`?

I was actually thinking outputting an empty string for all of them.  If an empty string was input, presumably that's the desired overall output?

Reply:
These are "conversion specifiers" they allow the input value to be converted to another type. While `D` converts to a string using `strftime()`, the `d` specifier converts to a datetime object, and the user can then use strftime format to get a custom formatting of that datetime object. If you return an empty string then it could cause the custom formatting to throw an exception (unless I'm mistaken)

See the [flux-jobs](https://flux-framework.readthedocs.io/projects/flux-core/en/latest/man1/flux-jobs.html#output-format) documentation for an example.

Reply:
@grondo yeah, became evident to me as i was adjusting tests